torch/csrc/cudnn/Conv.cpp

```diff
@@ -225,6 +225,14 @@ void cudnn_convolution_forward(
   size_t workspaceSize;
   CHECK(cudnnGetConvolutionForwardWorkspaceSize(handle, idesc.desc, wdesc.desc,
       cdesc.desc, odesc.desc, fwdAlg, &workspaceSize));
+  size_t memFree;
+  size_t memTotal;
+  cudaMemGetInfo(&memFree, &memTotal);
```

Comment:
Yes, but if it frees the blocks, and there's still not enough memory it has no other choice than to propagate the OOM. Also, in case there are no free blocks at the moment it will behave just as the regular allocator.